staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go

```diff
@@ -171,6 +175,11 @@ type handlerChainConfig struct {
 	proxyMux        *http.ServeMux
 	serviceLister   v1listers.ServiceLister
 	endpointsLister v1listers.EndpointsLister
+
+	// postProxyHandler keeps track of the handler that comes after the proxy so that it can be used
+	// to satisfy delegation cases.  It is set as a side-effect while building the handler chain.
+	// TODO refactor to run the proxy *after* authorization so we can simply run at the end of the handling chain
```

Comment:
> I'm all for belt and suspenders, but two belts?

If some clean simplification comes through later, I don't object to re-assessing it.  However, setting up a "sometimes skip authorization" path rubs me the wrong way.  I'm far more likely to support a "don't authorize at all" on the extension API server, but even then I'd recommend against it.